prow/jenkins/jenkins.go

```diff
@@ -292,3 +297,29 @@ func (c *Client) GetLog(job string, build int) ([]byte, error) {
 	}
 	return buf, nil
 }
+
+// Abort a Jenkins build that is either running or waiting in the queue.
+func (c *Client) Abort(job string, build *JenkinsBuild) error {
+	c.log("Abort", job, build)
+	u := fmt.Sprintf("%s/job/%s/%d/stop", c.baseURL, job, build.Number)
+	var header map[string]string
+	if build.IsEnqueued() {
+		// Openshift CI runs a Jenkins proxy that uses a local cache of job names
+		// to proxy requests to multiple Jenkins masters. In case there is a queued
+		// build that needs to be cancelled, the proxy has no idea what is the
+		// correct master to redirect to because the queue cancellation request bears
+		// no information about the job for which the build needs to be cancelled.
+		// The custom header is used by the proxy to figure the correct master out.
+		header = map[string]string{"Jenkins-Job": job}
+		u = fmt.Sprintf("%s/queue/cancelItem?id=%d", c.baseURL, build.ID)
+	}
+	resp, err := c.request(http.MethodPost, u, header)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return fmt.Errorf("response not 2XX: %s: (%s)", resp.Status, u)
```

Comment:
What other than 200 is acceptable?

Reply:
OK, may want to log non-200 in case there is some weird response we end up chasing in the future

Reply:
Yeah, this error is logged in the caller.